src/cpp/src/tokenizer.cpp

```diff
@@ -596,6 +600,37 @@ class Tokenizer::TokenizerImpl {
     std::string get_chat_template() {
         return m_chat_template;
     }
+
+private:
+     void read_vocab_from_detokenizer_model(const std::shared_ptr<ov::Model>& model) {
```

Comment:
It seems (hard to reason about that at glance and that's the point of the request) that the only reason this is a class function instead of stand alone one is that it needs access to `m_vocab`. If this is the case, `read_vocab_from_detokenizer_model()` should return the vocab and become stand alone function so it would be easier to judge what the function changes.

Reply:
Moved the function to utils.

Reply:
Move to tokenizer.cpp outside the class

Reply:
Can you return `vector` instead of passing by reference

Reply:
Done